lib/ReactViews/Map/Panels/SharePanel/SharePanel.jsx

```diff
@@ -40,6 +42,24 @@ const SharePanel = createReactClass({
         };
     },
 
+    advancedOptions() {
+        return this.state.advancedIsOpen;
+    },
+
+    toggleAdvancedOptions(e) {
+        if (this.advancedOptions()) {
+            this.setState({
+                advancedIsOpen: false
+            });
+        } else {
+            this.setState({
+                advancedIsOpen: true
+            });
+        }
+
+        this.forceUpdate();
```

Comment:
```js
this.setState((prevState) => ({ 
  advancedIsOpen: !prevState.advancedIsOpen 
});
```
is the recommended way to set new state based on previous state. Also, `setState` will cause an update unless your state values are mutable, so `forceUpdate` isn't needed.